### PR TITLE
Got `Either` to 100% coverage

### PR DESCRIPTION
**Pull request by AlparSzabados**
Improves: https://github.com/javaslang/javaslang/issues/1279

### CODE REVIEW

File: javaslang/src/test/java/javaslang/control/EitherTest.java [hunk 1]
```diff
@@ -83,6 +84,17 @@ public void shouldFoldRight() {
     }
 
     @Test
+    public void shouldReturnSameWhenCallingMapOnLeft() {
+        final Either<Integer, Object> actual = Either.left(1);
+        assertThat(actual.map(Object::toString)).isSameAs(actual);
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherTest.java]: is the method reference important here? If not, you could throw an exception, to make it clear that it's not called :)

File: javaslang/src/test/java/javaslang/control/EitherTest.java [hunk 2]
```diff
@@ -83,6 +84,17 @@ public void shouldFoldRight() {
...
+    }
+
+    @Test(expected = NoSuchElementException.class)
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherTest.java]: is there a symmetric test for this, i.e. `left(1).getRight()`?

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java [hunk 3]
```diff
@@ -169,6 +184,14 @@ public void shouldConvertRightProjectionOfRightToJavaOptional() {
         assertThat(Either.<Integer, String> right("1").right().toJavaOptional()).isEqualTo(Optional.of("1"));
     }
 
+    // -- transform()
+
+    @Test
+    public void shouldTransform() {
+        String transformed = of(42).transform(v -> String.valueOf(v.get()));
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: * please make local variables `final`, it's how @danieldietrich prefers them :)
* you could chose slightly simpler values, e.g. `of(1)`

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java [hunk 4]
```diff
@@ -66,8 +67,22 @@ public void shouldGetOnRightProjectionOfRight() {
 
     @Test
     public void shouldRightProjectionOrElseRightProjection() {
-        assertThat(Either.right(1).right().orElse(Either.right(2).right()).get()).isEqualTo(1);
-        assertThat(Either.<Integer, Integer> left(1).right().orElse(Either.<Integer, Integer> right(2).right()).get()).isEqualTo(2);
+        final RightProjection<Object, Integer> rightRight = Either.right(1).right();
+        final RightProjection<Object, Integer> leftRight = Either.<Object, Integer>left(1).right();
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: this should be simplified, like the other one above it

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java [hunk 5]
```diff
@@ -66,8 +67,22 @@ public void shouldGetOnRightProjectionOfRight() {
 
     @Test
     public void shouldRightProjectionOrElseRightProjection() {
-        assertThat(Either.right(1).right().orElse(Either.right(2).right()).get()).isEqualTo(1);
-        assertThat(Either.<Integer, Integer> left(1).right().orElse(Either.<Integer, Integer> right(2).right()).get()).isEqualTo(2);
+        final RightProjection<Object, Integer> rightRight = Either.right(1).right();
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: I think you could inline these two.
Also, `Either.right(1)` could more simply be written as `Right(1)` (with static import to `API`)

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java [hunk 6]
```diff
@@ -66,8 +67,22 @@ public void shouldGetOnRightProjectionOfRight() {
...
+        final RightProjection<Object, Integer> leftRight = Either.<Object, Integer>left(1).right();
+        final RightProjection<Object, Integer> rightProjection = Either.right(2).right();
+
+        assertThat(rightRight.orElse(rightProjection).get()).isEqualTo(1);
+        assertThat(leftRight.orElse(rightProjection).get()).isEqualTo(2);
+    }
+
+    @Test
+    public void shouldRightProjectionOrElseRightProjectionFromSupplier() {
+        final RightProjection<Object, Integer> rightRight = Either.right(1).right();
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: same, these two can be simplified and inlined

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java (hunk 3 above)
Comment (ruslansennov) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: 42 is the best value ever!

Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: It is indeed, but doesn't it have too much baggage for such a simple test :)?

File: javaslang/src/test/java/javaslang/control/EitherTest.java (hunk 2 above)
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/control/EitherTest.java]: Added test for `Left(1).get()` just to make it symmetric.

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java (hunk 3 above)
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: Made it final and changed number to `1`

File: javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java (hunk 6 above)
Comment (AlparSzabados) [inline javaslang/src/test/java/javaslang/control/EitherRightProjectionTest.java]: Done

File: javaslang/src/test/java/javaslang/control/EitherTest.java [hunk 7]
```diff
@@ -83,6 +87,24 @@ public void shouldFoldRight() {
...
+    public void shouldReturnSameWhenCallingMapOnLeft() {
+        final Either<Integer, Object> actual = Left(1);
+        assertThat(actual.map(v -> {
+            throw new IllegalStateException();
+        })).isSameAs(actual);
+    }
+
+    @Test(expected = NoSuchElementException.class)
+    public void shouldThrowIfRightGetLeft() {
+        assertThat(Right(1).getLeft());
```
Comment (paplorinc) [inline javaslang/src/test/java/javaslang/control/EitherTest.java]: you can drop the `assertThat` part, just `Right(1).getLeft();`

**Comment by paplorinc** [comment]
@codecov-io 
image

**Comment by danieldietrich** [comment]
@AlparSzabados Thank you!! 🥇

**Pull request merged**